pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/utilities/ProtocolData.java

```diff
@@ -82,19 +83,26 @@ public ProtocolData(Map<String, String> paramsMap) {
         parseTupleDescription();
 
         /*
-         * accessor - will throw exception from getPropery() if outputFormat is
+         * accessor - will throw exception if outputFormat is
          * BINARY and the user did not supply accessor=... or profile=...
-         * resolver - will throw exception from getPropery() if outputFormat is
+         * resolver - will throw exception if outputFormat is
          * BINARY and the user did not supply resolver=... or profile=...
          */
-        profile = getOptionalProperty("PROFILE");
+        profile = getUserProperty("PROFILE");
```

Comment:
should we preserve semantic of getUserProperty() that throws exception if the property is not present and getUserOptionalProperty() that returns null ?

Reply:
I used getUserProperty since I thought User property implicitly indicates that these are custom properties

Reply:
but you repeat the same code multiple times here -- throwing error if the property is not present, which makes it a mandatory "user property".